package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -0,0 +1,121 @@
+            n_frames. Iteration stops *before* this frame number,
+            which means that the trajectory would be read until the end.
+        step : int, optional
+            step between frames, default None becomes 1.
+
+        """
+        super(Ramachandran, self).__init__(atomgroup.universe.trajectory, **kwargs)
+        self.atomgroup = atomgroup
+
+    def _prepare(self):
+        self.residues = self.atomgroup.residues
+        res_min = np.min(self.atomgroup.universe.select_atoms("protein").residues)
+        res_max = np.max(self.atomgroup.universe.select_atoms("protein").residues)
+        if any([(residue < res_min or residue > res_max) for residue in self.residues]):
+            raise IndexError("Selection exceeds protein length")
+        elif any([residue == (res_min or res_max) for residue in self.residues]):
+            warnings.warn("Cannot determine phi and psi angles for the first or last residues")
+
+        self.phi_atoms = [residue.phi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+        self.psi_atoms = [residue.psi_selection() for residue in self.residues
+                          if res_min < residue < res_max]
+
+        self.angles = []
+
+    def _single_frame(self):
+        self.phi_angles = dihedral_calc(self.phi_atoms)
+        self.psi_angles = dihedral_calc(self.psi_atoms)
+        self.angles.append((self.phi_angles,self.psi_angles))
+
+    def _conclude(self):
+        self.angles = np.array(self.angles)
+
+    def plot(self, ax=None, color='k', marker='s', title=None):
+        """Plots data into standard ramachandran plot. Each time step in
+        self.angles is plotted onto the same graph.
+
+        Parameters
+        ----------
+        ax : :class:`matplotlib.axes.Axes`
+              If no `ax` is supplied or set to ``None`` then the plot will
+              be added to the current active axes.
+        color : string, optional
+              Color used for markers in the plot; the default color is 'black'.
+        marker : string, optional
+               Marker used in plot; the default marker is 'square'.
+        title : string, optional
+              Title of axes object; the default `None` leaves plot without a
+              title.
+
+        Returns
+        -------
+        ax : :class:`~matplotlib.axes.Axes`
+             Axes with the plot, either `ax` or the current axes.
+
+        """
+        if ax is None:
+            ax = plt.gca()
+        ax.axis([-180,180,-180,180])
+        ax.axhline(0, color='k', lw=1)
+        ax.axvline(0, color='k', lw=1)
+        ax.set_xticks(range(-180,181,60))
+        ax.set_yticks(range(-180,181,60))
+        ax.set_xlabel(r"$\phi$ (deg)")
+        ax.set_ylabel(r"$\psi$ (deg)")
+        if title is not None:
+            ax.set_title(title)
+        for angles in self.angles:
+            ax.plot(angles[0],angles[1], color=color,
```

Comment:
a better function would be `scatter` here. It doesn't need the line plot styling.

```python
ax.scatter(angles[0], angles[1], **kwargs)
```

Reply:
You can make this into a one-line and only call into the plotting routine once. Might make a speed difference if you have a lot of frames.

```python
a = self.angles.reshape((np.prod(self.angles.shape[:2]), 2))
ax.scatter(a[0], a[1], **kwargs)
```